pubmedcentral/download_and_convert_to_md.py

```diff
@@ -0,0 +1,179 @@
+import argparse
+import functools
+import multiprocessing as mp
+import os
+import re
+import shutil
+import subprocess
+import tarfile
+import traceback
+import xml.etree.ElementTree as ET
+
+from tqdm import tqdm
+
+from licensed_pile import logs
+from licensed_pile.scrape import get_page
+
+parser = argparse.ArgumentParser(description="Convert xml documents to markdown.")
+parser.add_argument("--filelist", help="The path to the filelist.txt file.")
+parser.add_argument(
+    "--output_dir", default="data/md/", help="Where the markdown files go."
+)
+parser.add_argument(
+    "--total_docs",
+    default=0,
+    type=int,
+    help="Total number of documents to convert, for debugging.",
+)
+parser.add_argument(
+    "--author_dir", default="data/authors/", help="Where the author files go."
+)
+parser.add_argument(
+    "--processes",
+    default=mp.cpu_count(),
+    type=int,
+    help="Number of processes to use for conversion.",
+)
+
+
+def get_authors(nxml_file: str, pmcid: str):
+    # get authors from nxml file
+    authors = []
+
+    tree = ET.parse(nxml_file)
+
+    # search for author tags
+    for author in tree.findall(".//contrib[@contrib-type='author']"):
+        surname = author.find("name/surname")
+        given_names = author.find("name/given-names")
+        if surname is not None and given_names is not None:
+            authors.append(f"{surname.text}, {given_names.text}")
```

Comment:
Given that we have it  split in the metadata lets store the authors as `List[{"first": str, "last": str}]`

Reply:
Done